test/SampleAggregatorTest.cpp

```diff
@@ -339,9 +339,29 @@ TEST_F(SampleAggregatorTest, epoch_application_total)
 
     EXPECT_DOUBLE_EQ(2.0, m_agg->sample_region(M_SIGNAL_TIME, reg_normal));
     EXPECT_DOUBLE_EQ(2.0, m_agg->sample_region(M_SIGNAL_TIME, GEOPM_REGION_HASH_UNMARKED));
-    // First epoch observed at step == 2, app finished at step == 4.  4 - 2 = 2
```

Comment:
Some comments in the new code would really help in understanding what your changes are doing.  It took us a while to decipher this thing last night.